Sources/DistributedActors/Pattern/WorkerPool.swift

```diff
@@ -12,18 +12,28 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Distributed
 import Logging
 
-/// A `WorkerPool` represents a pool of actors that are all equally qualified to handle incoming messages.
+public protocol DistributedWorker: DistributedActor {
```

Comment:
I'm on the fence if we should lock it down to the actor system:

```suggestion
public protocol DistributedWorker: DistributedActor where ActorSystem == ClusterSystem {
```

My gut feeling is that yes, that might be better for now.

Reply:
(There's an entire world of pain related to the "we need a concrete ActorSystem at all times" because of some language limitations -- by locking it down right away we'll avoid bumping into weird issues)

Reply:
Maybe you're right here though... keeping the protocol "wide open" but constraining when used with the pool... 🤔 

(Sorry, just thinking out loud -- just an option to consider if you'd hit some issues with this protocol and calling stuff on it)

Reply:
The pool relies on `ClusterSystem` features such as receptionist, logger, etc. so I suppose we do need the constraint?

Reply:
Yeah, it makes sense -- it's not a "general" feature as we need lifecycle watch especially.